src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Windows.cs

```diff
@@ -634,17 +634,37 @@ internal SerialStream(string portName, int baudRate, Parity parity, int dataBits
                 // set constant properties of the DCB
                 InitializeDCB(baudRate, parity, dataBits, stopBits, discardNull);
 
-                DtrEnable = dtrEnable;
-
-                // query and cache the initial RtsEnable value
-                // so that set_RtsEnable can do the (value != rtsEnable) optimization
-                _rtsEnable = (GetDcbFlag(Interop.Kernel32.DCBFlags.FRTSCONTROL) == Interop.Kernel32.DCBRTSFlowControl.RTS_CONTROL_ENABLE);
+                try
+                {
+                    DtrEnable = dtrEnable;
+                }
+                catch (IOException) when (dtrEnable == false)
```

Comment:
It is not a good approach to throw and immediately catch the exception.

Instead, the setter of `DtrEnable`/`RtsEnable` can be refactored into methods of `TryEnable`, then the callers here can call the `TryEnable` methods directly and ignores the failure.

Reply:
It's up to the area owners for decision. Following existing pattern is also acceptable.

Reply:
I'll leave it up to the area owners then.